### PR TITLE
Ctrl-Shift-Click points at stuff.

### PR DESCRIPTION
**Pull request by Fryer22**
-Ctrl-Shift-click points at whatever you clicked on, akin to right click -> object -> drop down -> point to. 

Details:
--A 20 second cooldown is of course attached to stop spam. 20 seconds is the amount of time the arrow lasts. Coincidentally, it is also the quickest I could actually do the previous method of pointing. Yes, I timed myself.
--AI's, MMI's/Brains, pAIs, and soul shards can't point. Obviously, people in disposals and lockers cannot as well.
--Cyborgs CAN point, just not at doors as borgs and AI's use ctrl shift click for access override. Borgs can still point at doors via the drop down menu. 
--You can't point at stuff through walls via x-ray or security cameras. Well, you can try, but no emote or arrow is created.
--To sum it up, if you could point at something via the drop down menu, you can point at it through this. 
--Bonus: Creates the possibility of having ~FUN~ additions such as a syndicate implant that makes you vomit a swarm of SyndBees (TM) to attack whoever you point at! Or just making slimes able to attack someone if you say "Kill" and point at the bartender. The possibilities are endless!

(Will add changelog after this is posted)

### CODE REVIEW

**Comment by vista-** [comment]
You mean control-shift-click.

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -253,8 +256,21 @@
 	A.CtrlShiftClick(src)
 	return
 
-/atom/proc/CtrlShiftClick(var/mob/user)
+/atom/proc/CtrlShiftClick(var/mob/living/user) 	//PybroPointClick, 1/30/15
+	if(istype(user, /mob/living/carbon/brain || /mob/living/silicon/pai)) //If this isn't in place, then they can still point. They won't actually make an arrow/emote, but they'll still get the tired finger message. Pedantic, but whatever.
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: No, this is not the right way to do this.

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -253,8 +256,21 @@
 	A.CtrlShiftClick(src)
 	return
 
-/atom/proc/CtrlShiftClick(var/mob/user)
+/atom/proc/CtrlShiftClick(var/mob/living/user) 	//PybroPointClick, 1/30/15
+	if(istype(user, /mob/living/carbon/brain || /mob/living/silicon/pai)) //If this isn't in place, then they can still point. They won't actually make an arrow/emote, but they'll still get the tired finger message. Pedantic, but whatever.
+		return
+	if(user.client && user.client.eye == user)	//Makes sure all of our seeing is in order.
+		if(user.pointcooldown == 1)				//This is to prevent spam.
+			user << "<span class='notice'>Give your finger a moment to cool down!</span>" //Cheesy, but whatever.
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: it's a little bit too cheesy.

File: code/_onclick/click.dm [hunk 3]
```diff
@@ -253,8 +256,21 @@
 	A.CtrlShiftClick(src)
 	return
 
-/atom/proc/CtrlShiftClick(var/mob/user)
+/atom/proc/CtrlShiftClick(var/mob/living/user) 	//PybroPointClick, 1/30/15
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: Remove this comment. the date isn't even in the right format either.

File: code/_onclick/click.dm [hunk 4]
```diff
@@ -253,8 +256,21 @@
...
+		return
+	if(user.client && user.client.eye == user)	//Makes sure all of our seeing is in order.
+		if(user.pointcooldown == 1)				//This is to prevent spam.
+			user << "<span class='notice'>Give your finger a moment to cool down!</span>" //Cheesy, but whatever.
+			return
+		else									//We point at it, face it, and get a tired finger.
+			user.pointed(src)
+			user.face_atom(src)
+			user.pointcooldown = 1
+			spawn(20)							//20 seconds is how long the arrow lasts. Consistency!
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: remove this comment, we can figure this out from the code itself

File: code/_onclick/click.dm [hunk 5]
```diff
@@ -253,8 +256,21 @@
 	A.CtrlShiftClick(src)
 	return
 
-/atom/proc/CtrlShiftClick(var/mob/user)
+/atom/proc/CtrlShiftClick(var/mob/living/user) 	//PybroPointClick, 1/30/15
+	if(istype(user, /mob/living/carbon/brain || /mob/living/silicon/pai)) //If this isn't in place, then they can still point. They won't actually make an arrow/emote, but they'll still get the tired finger message. Pedantic, but whatever.
+		return
+	if(user.client && user.client.eye == user)	//Makes sure all of our seeing is in order.
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: remove this comment, we can figure this out from the code itself

File: code/_onclick/click.dm [hunk 6]
```diff
@@ -253,8 +256,21 @@
 	A.CtrlShiftClick(src)
 	return
 
-/atom/proc/CtrlShiftClick(var/mob/user)
+/atom/proc/CtrlShiftClick(var/mob/living/user) 	//PybroPointClick, 1/30/15
+	if(istype(user, /mob/living/carbon/brain || /mob/living/silicon/pai)) //If this isn't in place, then they can still point. They won't actually make an arrow/emote, but they'll still get the tired finger message. Pedantic, but whatever.
+		return
+	if(user.client && user.client.eye == user)	//Makes sure all of our seeing is in order.
+		if(user.pointcooldown == 1)				//This is to prevent spam.
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: remove this comment, we can figure this out from the code itself

File: code/_onclick/click.dm [hunk 7]
```diff
@@ -253,8 +256,21 @@
...
 
-/atom/proc/CtrlShiftClick(var/mob/user)
+/atom/proc/CtrlShiftClick(var/mob/living/user) 	//PybroPointClick, 1/30/15
+	if(istype(user, /mob/living/carbon/brain || /mob/living/silicon/pai)) //If this isn't in place, then they can still point. They won't actually make an arrow/emote, but they'll still get the tired finger message. Pedantic, but whatever.
+		return
+	if(user.client && user.client.eye == user)	//Makes sure all of our seeing is in order.
+		if(user.pointcooldown == 1)				//This is to prevent spam.
+			user << "<span class='notice'>Give your finger a moment to cool down!</span>" //Cheesy, but whatever.
+			return
+		else									//We point at it, face it, and get a tired finger.
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: remove this comment, we can figure this out from the code itself

File: code/_onclick/click.dm [hunk 8]
```diff
@@ -253,8 +256,21 @@
...
+			user.pointed(src)
+			user.face_atom(src)
+			user.pointcooldown = 1
+			spawn(20)							//20 seconds is how long the arrow lasts. Consistency!
+				user.pointcooldown = 0			//Our finger recuperates.
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: remove this comment, we can figure this out from the code itself

File: code/_onclick/click.dm (hunk 2 above)
Comment (Fryer22) [inline code/_onclick/click.dm]: Any suggestions?

**Comment by RemieRichards** [comment]
Bit excessive with my line comments, sorry, but you get my point.

File: code/_onclick/click.dm (hunk 1 above)
Comment (Fryer22) [inline code/_onclick/click.dm]: Which part specifically? The //comment? The istype? The /atom/proc/etc?

Comment (RemieRichards) [inline code/_onclick/click.dm]: The istype()s you're placing Non Object Oriented Code in a proc specifically set up to be object oriented.

You should:
* Create a ```/mob/living/carbon/brain/CtrlShiftClick(var/mob/living/user)``` that just returns
* Create a ```/mob/living/silicon/pai/CtrlShiftClick(var/mob/living/user)``` that just returns

File: code/_onclick/click.dm (hunk 2 above)
Comment (RemieRichards) [inline code/_onclick/click.dm]: That's the issue, it's difficult to come up with one. if no better replacement comes along, this one can stay. or some variant on "your arm is resting" or such.

**Comment by Zelacks** [comment]
It seems this does not prevent you from spamming your point if you use the verb, which causes the shortcut to provide different functionality. Maybe subjective, but perhaps CtrlShiftClick(var/mob/user) should only contain the call to the pointed() function, as well as the point-to verb only containing a  call to pointed(), and all the logic should be done in pointed. This means that the shortcut and verb will always perform the same function, which is what I believe is ideal

File: code/_onclick/click.dm (hunk 4 above)
Comment (Miauw62) [inline code/_onclick/click.dm]: lol, the comment isn't even right, its 2 seconds, not twenty.

**Comment by Fryer22** [comment]
-Closing this due to a surprise of real life and fucking up the commits. I will polish and re-PR this when I get the chance. Sorry.
